Added note about updating ADB on Windows machines and format fixes

thatdocslady opened the pull request:
Fixes issue #500 

Added a note about additional steps required on Windows machines to update, and also restructured the Updating topic a little to accommodate for the new content and any future modular additions.

Note: I intentionally didn't list version numbers because that's a maintenance headache, and the only justification for that would be if the limitation is temporary, in which case it doesn't belong in the documentation but instead should be listed in the release notes.

brgnepal commented on docs/updating.rst:
```diff
@@ -2,35 +2,56 @@
 
-  3. If you see the above output, update the ADB Vagrant image with this
-     command::
+   Checking if box 'projectatomic/adb' is up to date...
```
@thatdocslady I think the lines from 21 to 24 should be in grey color as it is output of the `vagrant box outdated` command.

bexelbie commented:
@budhrg why does centos-ci get so chatty?

brgnepal commented:
@bexelbie Probably @dharmit could answer you well :smile:

bexelbie commented on docs/updating.rst:
```diff
@@ -2,35 +2,56 @@
+
+2. Run the following command to delete the `.vagrant.d` directory::
+
+   # rm -rf C:/Users/<user_name>/.vagrant.d/ 
```
is `rm` available in all recommended windows shells?

bexelbie commented:
@mariel, fyi - at this time there are no release notes for ADB - that needs to happen, but until then we need to be conscious of it in docs.  This doc appears to not be affected.

brgnepal commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
 
-  3. If you see the above output, update the ADB Vagrant image with this
-     command::
+      Checking if box 'projectatomic/adb' is up to date...
```
:+1:

thatdocslady commented:
@budhrg good catch about the output formatting, apparently this was already broken before I started working on it. Fixed it and also removed the redundant command, since it's already described in the previous step.

thatdocslady commented:
@bexelbie so should I list version numbers for this note? or can this step be useful / non-damaging for any Windows update procedures? better safe than sorry when it comes to updates, from my experience...

brgnepal commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
+--------------------------------------------------
 
-      # vagrant box outdated
+1. Check if a newer version of the box is available use this command::
```
@thatdocslady  I am not able to read this line properly as expected.

Do you think it should be:

```
To check if a new version of the box is available, use this command::
```

thatdocslady replied:
Right, will fix.

brgnepal commented:
LGTM except one suggestion.

thatdocslady commented:
@budhrg fixed the phrasing as you suggested

brgnepal commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
-     command::
+      Checking if box 'projectatomic/adb' is up to date...
+      A newer version of the box 'projectatomic/adb-testing' is available! You
+      currently have version '1.3.1'. The latest is version '1.3.2'. Run ``vagrant
```
@thatdocslady  Since we did formatting here, the quotes like "``" doesn't have any effect. Can we change to single quote?

Rest is looking fine :smile:

thatdocslady replied:
@budhrg done :)

brgnepal commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
+      Checking if box 'projectatomic/adb' is up to date...
+      A newer version of the box 'projectatomic/adb-testing' is available! You
+      currently have version '1.3.1'. The latest is version '1.3.2'. Run `vagrant
+      box update to update.
```
@thatdocslady 
Probably missed ending back tick after box update.
It should be 

```
Run `vagrant box update` to update. 
```

or

```
Run 'vagrant box update' to update.
```

thatdocslady commented:
@budhrg that's my punishment for trigger-happy committing ;) fixed now

thatdocslady commented:
@budhrg @bexelbie I think I fixed everything in the feedback, unless there are other comments can someone please merge this?

bexelbie commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
 
-  1. Check if a newer version of the box is available use this command::
+--------------------------------------------------
+Update the ADB for images from atlas.hashicorp.com
```
Should this be "Updating"?

It reads funny to me.  Perhaps "Update the ADB using atlas.hashicorp.com" ?

thatdocslady replied:
@bexelbie I try to avoid gerunds in task titles, but maybe that's a personal/historical style preference.

Maybe "Update ADB from atlas.h..."?

bexelbie commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
+    # vagrant box update
 
-      # vagrant up
+4. Once the image is updated, you can restart the ADB with the new version
```
This needs to be "start" not "restart", I think (see above)

bexelbie commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
-* If you are using the image from atlas.hashicorp.com
+Depending on your implementation of the ADB, you might be able to directly
+update your Vagrant box. In some cases, you will need to perform a clean
+installation or follow additional steps to complete the update.
```
We need a prerequisite statement that says that you must not only `vagrant halt` your existing ADB, but you must actually `vagrant destroy` it.

This will lose any data or configuration in your ADB that is not saved on your host.

@budhrg please confirm.

thatdocslady replied:
@bexelbie @budhrg if you're already discussing this prereq, can you clarify whether it's required only for this image source and at which point to run the command? thx

brgnepal replied:
@thatdocslady @bexelbie :
Yes clean installation requires `vagrant destroy` and we should add a warning that this step will potentially remove any data or configuration the user performed with current ADB VM and it is the user's decision whether to update the Vagrant box or not.

thatdocslady replied:
@budhrg so where should I add this step? Is it relevant only for the centos images? As it looks now, the atlas images can be directly updated without a clean install.

thatdocslady replied:
It looks like Naina confirmed that this step is required for CDK, which means only CentOS images. Will adjust accordingly.

LalatenduMohanty commented on docs/updating.rst:
```diff
@@ -2,35 +2,55 @@
+install as described in the `installation documentation <installing.rst>`_.
+
+-------------------------
+Note for Windows machines
```
This below section is not for ADB but for updating Vagrant version in Windows. I do not think this section should be part of how to update ADB documentation. This kind of documentation should be part of a separate documentation.

thatdocslady commented:
@bexelbie @budhrg restructured the topic and added the vagrant destroy prerequisites, please review :)

brgnepal commented on docs/updating.rst:
```diff
@@ -1,36 +1,53 @@
 
-      # vagrant box outdated
+If you cannot directly update your Vagrant box, you must perform the following steps
+to remove the previously-installed boxes from your virtual machine.
```
IMO, "previously-installed boxes from your virtual machine." should be "previously-installed boxes from your host machine."

bexelbie replied:
+1

brgnepal commented on docs/updating.rst:
```diff
@@ -1,36 +1,53 @@
+to remove the previously-installed boxes from your virtual machine.
 
-  2. If a newer version is available, the output should look like this::
+1. If ADB is running, run the `vagrant halt` command.
```
Some indentation issue here 
![screenshot from 2016-09-01 20-19-25](https://cloud.githubusercontent.com/assets/1132451/18171971/6c3f2d52-7081-11e6-8a25-956d79e076e9.png)

brgnepal replied:
Also note that "vagrant halt" is not visible in code block. I believe it should be wrapped inside double back ticks (rst format)

brgnepal commented on docs/updating.rst:
```diff
@@ -1,36 +1,53 @@
+1. If ADB is running, run the `vagrant halt` command.
 
-        # vagrant box outdated
+2. Delete the Vagrant boxes with the `vagrant destroy` command. This action will also
```
"vagrant destroy" will remove "virtual machine". Please update the sentence as "Remove the Virtual machine with 'vagrant destroy' command.

bexelbie replied:
I don't think you need this line and I don't think `vagrant box update` does anything similar.  What will happen is that a new box will be added to the ~/.vagrant directory.  Boxes are versioned so they don't clobber each other.

That said, a note that suggests that removing the box with the above command along with a --version string (or something like that check vagrant help) to free up disk space may be useful.  However, I can see omitting this on the presumption that our job isn't to really teach vagrant ... dunno.

brgnepal replied:
Ya agreed that these steps looking like we are teaching vagrant here :wink: .

IMO there are two ways how these boxes are managed by developers usually.
- Automatic download , update:
  
  If we do `vagrant up` on Vagrantfile where box name describe in `config.vm.box` is not present locally, then vagrant will download box from "atlas.hashicorp.com". This will also allow automatic update with `vagrant box update`.
- Download box, add box:
  
  Second way is to download the box like anywhere from `cloud.centos.org`, then add through `vagrant box add <box_name> <local box url / remote url>` command and specify this `box_name` in Vagrantfile and does "vagrant up". In this case, automatic update is not possible. Then updating will need download new updated box, add that box. If same name is to be used then probably --version is need to be used otherwise user will get error as box name already exists. Otherwise, user need to remove the box and then add updated box.

Hope you will get some insights on how users generally use.

brgnepal commented on docs/updating.rst:
```diff
@@ -1,36 +1,53 @@
+1. If ADB is running, run the `vagrant halt` command.
 
-        # vagrant box outdated
+2. Delete the Vagrant boxes with the `vagrant destroy` command. This action will also
```
One more step is required to delete the Vagrant box (as in case of `vagrant box update` it will be done automatically) as:

```
vagrant box remove  projectatomic/adb [or box name]
```

brgnepal commented on docs/updating.rst:
```diff
@@ -1,36 +1,53 @@
-     A newer version of the box 'projectatomic/adb-testing' is available! You
-     currently have version '1.3.1'. The latest is version '1.3.2'. Run ``vagrant
-     box update`` to update.
+3. If you are updating ADB on a Windows machine, run the following command to delete
```
I think we can specify this point as Note. Since it's a special case.

bexelbie replied:
At a minimum, we need to call it out above.

brgnepal commented:
@thatdocslady I have reviewed from my side.

All good except few suggestions. Please check.

bexelbie commented on docs/updating.rst:
```diff
@@ -1,36 +1,53 @@
+-------------
 
-      # vagrant box outdated
+If you cannot directly update your Vagrant box, you must perform the following steps
```
How do I know if I can directly update or not?

brgnepal replied:
Updating the sentence with hyperlink to the section "Update ADB from atlas.hashicorp.com" should work.

thatdocslady commented:
@bexelbie @budhrg I think I may have been able to address all your points, did some more restructuring and I feel that the flow of the content is much more logical now for each use-case.

Also, unfortunately admonitions (notes, warnings) in RST and ASCIIDoc are intentionally not live-rendered in GitHub[1], and since the Windows machine step must be performed before you download the new image it's probably better to keep it as a step in the task and the fact that we specify Windows machines is enough IMO.

[1] https://github.com/github/markup/issues/68

brgnepal commented:
@thatdocslady LGTM :+1:

thatdocslady commented:
@bexelbie can you please review and merge if LGTY? :)

thatdocslady commented:
@bexelbie took out the prerequisites to their own section based on @LalatenduMohanty 's needed info, please review and merge if ok

bexelbie commented on docs/updating.rst:
```diff
@@ -1,36 +1,63 @@
+   delete the `.vagrant.d` directory::
 
-      # vagrant box update
+     # rm -rf C:/Users/<user_name>/.vagrant.d/
```
I'd like to see the response to https://github.com/projectatomic/adb-atomic-developer-bundle/issues/500#issuecomment-244985560 incorporated here - this feels like a cloud.centos edge case

thatdocslady replied:
@bexelbie see https://github.com/projectatomic/adb-atomic-developer-bundle/issues/500#issuecomment-247329072, naina confirmed this is CDK (which means centos), I'll move the step to that task and rebase.

thatdocslady commented:
@bexelbie @LalatenduMohanty moved the Windows comment to the section for centos images based on Naina's comment. Please review and merge if ok.

bexelbie commented on docs/updating.rst:
```diff
@@ -1,36 +1,63 @@
+1. Monitor the mailing list and other announcement points to determine if a new image
+   is available.
+
+2. If you are updating ADB on a Windows machine, run the following command to
```
I still think this should be in a note that mentions in case you run into trouble.  I don't think this is a required step for every upgrade.

thatdocslady replied:
@bexelbie we had this discussion before already and I explained that if someone needs to do this step it must be in order, and it's better to sign-post it as a preventive rather than a "if you fail, go back and do this". It's not destructive as much as vagrant destroy, and it's better to run it anyway from what I understand.

bexelbie commented:
LGTM

brgnepal commented:
+1

Merged. Thanks for the PR.
